lib/aws/util.ex

```diff
@@ -5,7 +5,7 @@ defmodule AWS.Util do
   Create an HMAC-SHA256 for `key` and `message`.
   """
   def hmac_sha256(key, message) do
-    :crypto.hmac(:sha256, key, message)
+    :crypto.mac(:hmac, :sha256, key, message)
```

Comment:
We need to support the older versions of OTP for a while. So my suggestion is to conditionally define the function based on `System.otp_release()`.
It would be something like this:

```elixir
  if System.otp_release() >= "24" do
    def hmac_sha256(key, message) do
      :crypto.mac(:hmac, :sha256, key, message)
    end
  else
    def hmac_sha256(key, message) do
      :crypto.hmac(:sha256, key, message)
    end
  end
```
WDYT?

Reply:
Yeah agreed, would you prefer `>= "22"` or `>= "24"`?

I had a look at [plug_crypto](https://github.com/elixir-plug/plug_crypto/pull/20/files) and they've used the same approach but with a version check of 22

Reply:
I think "22" is better :)